libraries/net/message_oriented_connection.cpp

```diff
@@ -251,6 +255,7 @@ namespace graphene { namespace net {
         }
         ~verify_no_send_in_progress() { var = false; }
       } _verify_no_send_in_progress(_send_message_in_progress);
```

Comment:
Since the `assert` above will be skipped in release build, the `elog` above doesn't make sense. If we want to avoid two tasks calling `send_message` at the same time, we need a lock.

There is another `elog` below which looks suspicious as well.

Reply:
I haven't seen any indication that this happens (i. e. two tasks calling `send_message` in parallel), so I wouldn't change this at this time.

The check below (`MAX_MESSAGE_SIZE`) is OK IMO. We will start sending the oversized message, but the remote end will probably disconnect us. If they don't it's fine too.
Note that a too large message is an indication of a problem in the `chain` layer - it should catch oversized blocks and transactions. Protocol messages are much smaller than the limit.

Hm, actually... this should be verified for some message types. But out of scope here.